### PR TITLE
Add initial ParallelBus support for ESP32-S2

### PR DESCRIPTION
**Pull request by kmatch98**
I updated and commited a new revision that gives more pin options.  See note below....  This is ready for review.

#### Special Note: 
Due to what I think is a limitations of the ESP32-S2 that prevents writing a  single byte (8-bits) to the output register (32-bits, I had to modify the code to write a full 32-bit chunk to the register. **I would like feedback if there are faster methods for writing out the data (see lines 214-236).**

(Note: If using the same register for the data lines and the write line, so I can clear the write line at the same time as writing the data.  I verified that I could eliminate the "clear_write" line with no negative impact on the operation, but with approximately 13% total time reduction.)

#### Verified on the following hardware:
- ESP32-S2 Saola Wrover demo board ([ESP32-S2 Saola 1R Dev Kit featuring ESP32-S2 WROVER, Adafruit product ID: 4693](https://www.adafruit.com/product/4693))
- 320x240 LCD display using ILI9341 display driver ([2.8" TFT LCD with Touchscreen Breakout Board w/MicroSD Socket - ILI9341, Adafruit product ID: 1770](https://www.adafruit.com/product/1770))
- CircuitPython version 6.1.0: `6.1.0-rc.1-15-g6a76b6002-dirty on 2021-01-22; Saola 1 w/Wrover with ESP32S2`

#### Benchmark results

200x writes of [purple.bmp](https://cdn-learn.adafruit.com/assets/assets/000/074/681/original/purple.bmp?1555717198) using a forced refresh `display.refresh()` for each write:
- [PyPortal](https://www.adafruit.com/product/4116?gclid=Cj0KCQiAjKqABhDLARIsABbJrGkAVTE7amRHseazCPWOydHt2pXRW0311BO9A3O0my6IA4BlMOIWA3IaAlRAEALw_wcB): 34.9 seconds (running `6.0.0-alpha.3-33-gd600759bc-dirty on 2020-09-10; Adafruit PyPortal with samd51j20`)
- [Saola ESP32-S2](https://www.adafruit.com/product/4693): 23.3 seconds (running `6.1.0-rc.1-15-g6a76b6002-dirty on 2021-01-22; Saola 1 w/Wrover with ESP32S2`

### CODE REVIEW

**Comment by kmatch98** [comment]
I evaluated the impact of bit manipulation to convert the 8-bit data into the 32-bit register.  For everything I tried, I saw no significant impact on the performance.  If you have any recommendations on speeding it up, I'll be glad to try them.

**Comment by kmatch98** [comment]
I added capability for data and write pins >= pin #32.  Also, the data and write pins can be in different registers (0-31, or 32+).  

**This is now ready for review.**  Best is to compare against the [SAMD](https://github.com/adafruit/circuitpython/blob/main/ports/atmel-samd/common-hal/displayio/ParallelBus.c) or [NRF](https://github.com/adafruit/circuitpython/blob/main/ports/nrf/common-hal/displayio/ParallelBus.c) ports.

------

#### I verified on the following hardware:
- ESP32-S2 Saola Wrover demo board (ESP32-S2 Saola 1R Dev Kit featuring ESP32-S2 WROVER, Adafruit product ID: 4693)
- 320x240 LCD display using ILI9341 display driver (2.8" TFT LCD with Touchscreen Breakout Board w/MicroSD Socket - ILI9341, Adafruit product ID: 1770)
- CircuitPython version 6.1.0: 6.1.0-rc.1-15-g6a76b6002-dirty on 2021-01-22; Saola 1 w/Wrover with ESP32S2

#### Tests:
I verified the use of data0 pins in the lower register (pins #8-15), with either write pin #3 or write pin #33. 
**Still unverified:**
I don't have hardware that breaks out pin 32, so I could not verify operation with data0 pins in the higher register (32+).

File: ports/esp32s2/common-hal/displayio/ParallelBus.c [hunk 1]
```diff
@@ -33,35 +33,211 @@
...
+ *
+ */
+
 void common_hal_displayio_parallelbus_construct(displayio_parallelbus_obj_t* self,
     const mcu_pin_obj_t* data0, const mcu_pin_obj_t* command, const mcu_pin_obj_t* chip_select,
     const mcu_pin_obj_t* write, const mcu_pin_obj_t* read, const mcu_pin_obj_t* reset) {
 
-    mp_raise_NotImplementedError(translate("ParallelBus not yet supported"));
+    uint8_t data_pin = data0->number;
+    if ( (data_pin % 8 != 0) && (data_pin >= 32) ) {
```
Comment (tannewt) [inline ports/esp32s2/common-hal/displayio/ParallelBus.c:48]: What is the second part of the check doing? Seems like you should only need the first part.

File: ports/esp32s2/common-hal/displayio/ParallelBus.c [hunk 2]
```diff
@@ -33,35 +33,211 @@
...
+	}
+
+
+    mp_printf(&mp_plat_print, "write_clear: %x, write_set: %x\n", self->write_clear_register, self->write_set_register);
+
+    self->write_mask = 1 << (write->number % 32); /* the write pin triggers the LCD to latch the data */
+    mp_printf(&mp_plat_print, "write_mask: %x\n", self->write_mask);
+
+    mp_printf(&mp_plat_print, "out1 register: %x\n", g->out1.val);
+    mp_printf(&mp_plat_print, "clear a bit\n");
```
Comment (tannewt) [inline ports/esp32s2/common-hal/displayio/ParallelBus.c:122]: Please remove these prints now that it's working.

**Review by tannewt — CHANGES_REQUESTED** [review]
Just one question and a request. Looks good overall. Thanks for taking this on!

File: ports/esp32s2/common-hal/displayio/ParallelBus.c (hunk 2 above)
Comment (kmatch98) [reply ports/esp32s2/common-hal/displayio/ParallelBus.c]: Deleted and re-committed.  Thanks for the feedback.  I need to put "check for printf" on my pre-commit checklist!

**Review by tannewt — APPROVED** [review]
Thank you!

**Pull request merged**